contract/src/test/java/io/neow3j/contract/TransactionBuilderTest.java

```diff
@@ -84,137 +86,254 @@ public void setUp() {
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
+        try {
+            Long nonce = Integer.toUnsignedLong(-1) + 1;
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+
+        try {
+            Long nonce = (long) Math.pow(2, 32);
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+
+        try {
+            Long nonce = -1L;
+            b.nonce(nonce);
+            fail();
+        } catch (TransactionConfigurationException ignored) {
+        }
+    }
+
+    @Test(expected = TransactionConfigurationException.class)
+    public void failBuildingTransactionWithNegativeValidUntilBlockNumber() {
+        new TransactionBuilder(neow)
+                .validUntilBlock(-1L)
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
+    }
+
+    @Test(expected = TransactionConfigurationException.class)
+    public void failBuildingTransactionWithTooHighValidUntilBlockNumber() {
+        new TransactionBuilder(neow)
+                .validUntilBlock((long) Math.pow(2, 32))
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()));
     }
 
     @Test
-    public void testAutomaticSettingOfValidUntilBlockVariable() throws IOException {
-        Wallet wallet = Wallet.createWallet();
-        setUpWireMockForCall("invokescript", "invokescript_name_neo.json", SCRIPT);
-        ContractTestHelper.setUpWireMockForGetBlockCount(1000);
+    public void automaticallyFetchValidUntilBlock() throws Throwable {
+        setUpWireMockForCall("getblockcount", "getblockcount_1000.json");
+        setUpWireMockForCall("invokescript", "invokescript_transfer.json");
 
-        Invocation i = new Invocation.Builder(neow)
-                .withScript(Numeric.hexStringToByteArray(SCRIPT))
-                .withWallet(wallet)
-                .build();
-        assertThat(i.getTransaction().getValidUntilBlock(),
-                is((long) NeoConstants.MAX_VALID_UNTIL_BLOCK_INCREMENT + 1000 - 1));
+        Transaction transaction = new TransactionBuilder(neow)
+                .wallet(Wallet.withAccounts(account1))
+                .script(new byte[]{1, 2, 3})
+                .signers(Signer.calledByEntry(account1.getScriptHash()))
+                .buildTransaction();
+
+        assertThat(transaction.getNonce(), greaterThanOrEqualTo(0L));
+        assertThat(transaction.getNonce(), lessThanOrEqualTo((long) Math.pow(2, 32)));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void failBuildingTxWithoutAnySigner() throws Throwable {
```

Comment:
In my opinion this test should be made more thorough by using the `exceptionRule` and looking for specific key words in the exception method, because the IllegalStateException can arise from different things when calling `buildTransadtion()`.